threes.py

```diff
@@ -6,15 +6,20 @@
 
 def to_val(x):
     x = np.asarray(x)
-    return np.where(x < 3, x, 3*2**(x-3))
+    # integer powers to negative integers break as of python 3.5
+    # Ref: https://github.com/numpy/numpy/issues/8917
+    # Create a version of the array with no negative integers
+    min0 = np.where(x < 3, 0, x-3)
+    return np.where(x < 3, x, 3*2**(min0))
```

Comment:
We can simplify this to `3*2**np.max(0, x-3)` I think, which is clearer to me. The long comment is also unnecessary; the comment should be placed in the git log instead.